### DISCUSSION TITLE
Dynamic route with @symbol ? Similar to "mywebsite.com/@username"

### DISCUSSION THREAD

**Discussion by ahetawal-p** [vercel/next.js#43898] [Help] [answered]
Hello,
I have been following this discussion here, as its very similar. 
https://github.com/vercel/next.js/discussions/31932

I am trying to achieve a different result though however when I use the below redirect mentioned.

```
rewrites: async () => {
    return [
      {
        source: "/@:username",
        destination: "/profile/@:username",
      },
    ];
  },
```

Using above I see: `http://localhost:3080/@tester` => `http://localhost:3080/profile/tester` (missing `@` symbol here)

But I am looking for it `http://localhost:3080/@tester` => `http://localhost:3080/profile/@tester`  (with `@` symbol in the browser)

Having a consistent `@` in browser is important for us to make it seamless redirect. 

So wondering is it possible on server side, or I have to resort to useEffect and `router.replace()` to achieve this.

**Comment by wwwdepot** [comment, marked as answer]
Use middleware when dealing with complex redirections or rewrites

```tsx
import { NextResponse, URLPattern } from "next/server";
import type { NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  const pattern = new URLPattern("/@:username", req.url);
  const username = pattern.exec(req.url)?.pathname?.groups?.username;
  if (username) {
    // Do rewrite/redirect here
    return NextResponse.redirect(new URL(`/profile/@${username}`, req.url));
  }
}
```
